src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -318,27 +294,60 @@ private void CreateRequestBuilderClass(CodeNamespace currentNamespace, OpenApiUr
-            };
-            currentPathProperty.Type = new CodeType(currentPathProperty) {
-                Name = "string",
-                IsExternal = true,
-            };
-            currentClass.AddProperty(currentPathProperty);
-
             var httpCoreProperty = new CodeProperty(currentClass) {
-                Name = "httpCore",
-                Description = "Core service to use to execute the requests",
-                PropertyKind = CodePropertyKind.HttpCore
+                Name = httpCoreParameterName,
+                Description = "The http core service to use to execute the requests.",
+                PropertyKind = CodePropertyKind.HttpCore,
+                Access = AccessModifier.Private,
+                ReadOnly = true,
             };
             httpCoreProperty.Type = new CodeType(httpCoreProperty) {
                 Name = coreInterfaceType,
                 IsExternal = true,
+                IsNullable = false,
             };
             currentClass.AddProperty(httpCoreProperty);
+            var constructor = currentClass.AddMethod(new CodeMethod(currentClass) {
+                Name = constructorMethodName,
+                MethodKind = isRootClientClass ? CodeMethodKind.ClientConstructor : CodeMethodKind.Constructor,
+                IsAsync = false,
+                IsStatic = false,
+                Description = $"Instantiates a new {currentClass.Name} and sets the default values.",
+                Access = AccessModifier.Public,
+            }).First();
+            constructor.ReturnType = new CodeType(constructor) { Name = voidType, IsExternal = true };
+            if(isRootClientClass) {
+                constructor.SerializerModules = config.Serializers;
+                constructor.DeserializerModules = config.Deserializers;
+            } else {
+                var currentPathProperty = new CodeProperty(currentClass) {
+                    Name = currentPathParameterName,
+                    Description = "Current path for the request",
+                    PropertyKind = CodePropertyKind.CurrentPath,
+                    Access = AccessModifier.Private,
+                    ReadOnly = true,
+                };
+                currentPathProperty.Type = new CodeType(currentPathProperty) {
+                    Name = "string",
+                    IsExternal = true,
+                    IsNullable = false,
+                };
+                currentClass.AddProperty(currentPathProperty);
+                constructor.AddParameter(new CodeParameter(constructor) {
+                    Name = currentPathParameterName,
+                    Type = currentPathProperty.Type,
+                    Optional = false,
+                    Description = currentPathProperty.Description,
+                    ParameterKind = CodeParameterKind.CurrentPath,
+                });
+            }
+            constructor.AddParameter(new CodeParameter(constructor) {
```

Comment:
I won't add samples as comments in the code, this would get messy really quick, this is why we have the samples repo and why I always join an update whenever things change. Unit tests should also help decipher the code (which I though was self-explanatory).

I can however add doc comments on things like enum values (methodkin, parameterkind...) to clarify the intent of each specialized version of a code element. Thoughts?

Additionally, the doc comments added with #324  should help clarify a lot of the expectations (although that was intended for library implementers, it'll also help people working on other languages)